psh/test-date.py

```diff
@@ -0,0 +1,197 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh date command test
+#
+# Copyright 2022, 2023 Phoenix Systems
+# Author: Mateusz Niewiadomski, Mateusz Bloch
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import psh.tools.psh as psh
+
+DFLT_DATE_REG = r"\w{3}, \d{2} \w{3} (?:\d{2}|\d{4}),? \d{2}:\d{2}:\d{2}(?:\w{3}|\w{4})?\r+\n"
+SPEC_DATE_REG = (
+    r"Sun, 13 Sep (?:20|2020),? 12:\d{2}:\d{2}(?:\w{3}|\w{4})?\r+\n"  # arbitrary deta of epoch 1600000000 GMT
+)
+
+
+# Utility functions
+def invalid_format(format):
+    return f"date: invalid format '{str(format)}'"
+
+
+def invalid_date(format):
+    return f"date: invalid date '{format}'"
+
+
+def unrec_arg(arg):
+    return f"Unrecognized argument: {arg}"
+
+
+# Test parts
+def test_corrects(p):
+    """Testing correct `date` commands"""
+
+    # Test help
+    psh.assert_cmd(p, "date -h", expected=r"Usage(.*?)FORMAT(.*?)\r+\n", is_regex=True)
+
+    # Test printing and formatting
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+    psh.assert_cmd(p, "date +%Y", expected=r"\b(19|20)\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date +%H:%M:%Sformat", expected=r"\d{2}:\d{2}:\d{2}format\r+\n", is_regex=True)
+
+    # Test setting to value
+    psh.assert_cmd(p, "date -s @1600000000", expected=SPEC_DATE_REG, is_regex=True)
+    psh.assert_cmd(p, "date", expected=SPEC_DATE_REG, is_regex=True)
```

Comment:
I'd leave only the following pattern (it will be flexible enough IMHO):
`12:26:4\d`

I assume it won't take more than 9s to get an output of the next `date` call.